### PR TITLE
Always add unlabeled category for scanvi

### PR DESCRIPTION
**Pull request by adamgayoso**
Even when the labels don't contain the unlabeled category. This fixes an edge case with scarches (see added test case).

I also cleared some unused code.

### CODE REVIEW

File: scvi/data/anndata/_utils.py [hunk 1]
```diff
@@ -146,16 +145,6 @@ def _make_obs_column_categorical(
         )
     adata.obs[alternate_column_key] = codes
 
-    if not return_mapping:
```
Comment (jjhong922) [inline scvi/data/anndata/_utils.py:149]: thanks

**Review by jjhong922 — APPROVED** [review]
looks great, really simplifies a lot of code

File: scvi/data/anndata/fields/_scanvi.py [hunk 2]
```diff
@@ -48,24 +48,25 @@ def _remap_unlabeled_to_final_category(
...
-                self.attr_key,
-                categorical_dtype=cat_dtype,
-                return_mapping=True,
-            )
-            remapped = True
-        else:
-            remapped = False
+        # could be in mapping in transfer case
+        elif self._unlabeled_category not in mapping:
+            # no unlabeled category, so no need to remap
```
Comment (adamgayoso) [inline scvi/data/anndata/fields/_scanvi.py:53]:
```suggestion
```
